src/transformers/models/trillsson_efficient/feature_extraction_trillsson_efficient.py

```diff
@@ -0,0 +1,298 @@
+        self.f_max = f_max
+        self.f_min = f_min
+        self.fft_length = fft_length
+
+    @classmethod
+    def pad_symmetric(cls, raw_speech: np.ndarray, padding: int):
+        """Symmetric pad a 2D Tensor"""
+        if raw_speech.ndim != 2:
+            raise ValueError(f"Raw speech must be rank 2 got {raw_speech.shape}")
+        raw_speech_len = raw_speech.shape[1]
+        if padding > raw_speech_len:
+            pad_raw_speech = cls.repeat_n_times_with_extra(raw_speech, padding, raw_speech_len)
+        else:
+            pad_raw_speech = np.pad(raw_speech, [(0, 0), (0, padding)], mode="symmetric")
+        return pad_raw_speech
+
+    @classmethod
+    def pad_waveform(cls, raw_speech: np.ndarray, padding: int, mode: str, padding_value: float):
+        """Pad waveform"""
+        if raw_speech.ndim != 2:
+            raise ValueError(f"Raw speech must be rank 2 got {raw_speech.shape}")
+        if mode == "symmetric":
+            return cls.pad_symmetric(raw_speech, padding)
+        else:
+            pad_raw_speech = np.pad(raw_speech, [(0, 0), (0, padding)], mode=mode, constant_values=padding_value)
+            return pad_raw_speech
+
+    @staticmethod
+    def repeat_n_times_with_extra(raw_speech: np.ndarray, padding: int, raw_speech_len: int):
+        """Pad symmetric longer than the original raw speech"""
+        if raw_speech.ndim != 2:
+            raise ValueError(f"Raw speech must be rank 2 got {raw_speech.shape}")
+        num_copies = np.floor_divide(padding, raw_speech_len)
+        r = np.fliplr(raw_speech)
+        f = np.concatenate((r, raw_speech), axis=1)
+        copies = np.tile(f, [1, np.floor_divide(num_copies, 2)])
+        if num_copies % 2 != 0:
+            copies = np.concatenate((copies, r), axis=1)
+        pre_pad = np.concatenate([raw_speech, copies], axis=1)
+        extra = padding % raw_speech_len
+        pad_raw_speech = np.pad(pre_pad, [(0, 0), (0, extra)], mode="symmetric")
+        return pad_raw_speech
+
+    @staticmethod
+    def stabilized_log(data: np.ndarray, floor: float, additive_offset: float):
+        """Stabilized MelSpectrogram"""
+        return np.log(np.maximum(data, floor) + additive_offset)
+
+    @staticmethod
+    def rescale(input_array: np.ndarray, rate: float = 1.0 / 128.0, offset: int = -1):
+        """Rescale the input array"""
+        return input_array * rate + offset
+
+    def log_mel_spectrogram(
+        self,
+        raw_speech: np.ndarray,
+        audio_sample_rate: int,
+    ):
+        """Convert waveform to LogMelSpectrogram"""
+        window_length_samples = int(round(audio_sample_rate * self.window_length_secs))
+        hop_length_samples = int(round(audio_sample_rate * self.hop_length_secs))
+        if self.fft_length:
+            fft_length = self.fft_length
+        else:
+            fft_length = 2 ** int(np.ceil(np.log(window_length_samples) / np.log(2.0)))
+        spectrogram = tf.abs(
+            tf.signal.stft(
```

Comment:
We can compute STFT features in NumPy!

Reply:
Yep! We added this with whisper, see [here](https://github.com/huggingface/transformers/blob/main/src/transformers/models/whisper/feature_extraction_whisper.py#L171)